rules/rules.js

```diff
@@ -96,10 +122,13 @@ const getGroupsForItem = function (ruleConfig) {
  * @param {String} ruleConfig.description a description of the rule
  * @param {*} ruleConfig.execute callback that will be called when the rule fires
  * @param {HostTrigger|HostTrigger[]} ruleConfig.triggers triggers which will define when to fire the rule
+ * @param {String} ruleConfig.id the UID of the rule
+ * @param {String} ruleConfig.ruleGroup the name of the rule group to use.
  * @returns {HostRule} the created rule
  */
 let JSRule = function (ruleConfig) {
     let ruid = ruleConfig.id || ruleConfig.name.replace(/[^\w]/g, "-") + "-" + utils.randomUUID();
+    removeRule(ruid);
```

Comment:
I'm not sure that it's the best thing to _force_ prior removal. This means that if a script relies on failure if it tries to insert a rule a second time, it will change it's behaviour. More concretely: if two rules with the same uid are inserted, then the existing behaviour is that the second will fail. The new behaviour is that the first will be silently overwritten.

My preference would be to (a) retain the original behaviour by default (as overwriting is not a common use-case, I'd expect that doing it accidentally is more common and failing is good in this case) and (b) allow explicit overwriting by setting a parameter.